whois-api/src/main/java/net/ripe/db/whois/api/httpserver/HttpTransportRule.java

```diff
@@ -0,0 +1,45 @@
+package net.ripe.db.whois.api.httpserver;
+
+import org.eclipse.jetty.rewrite.handler.Rule;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Enumeration;
+
+public class HttpTransportRule extends Rule {
+
+    private final String transport;
+    private final Rule delegate;
+
+    public HttpTransportRule(final String transport, final Rule delegate) {
+        this.transport = transport;
+        this.delegate = delegate;
+    }
+
+    @Override
+    public String matchAndApply(String target, HttpServletRequest request, HttpServletResponse response) throws IOException {
+        return hasTransport(request)? delegate.matchAndApply(target, request, response) : null;
+    }
+
+
+    private boolean hasTransport(final HttpServletRequest request) {
+        final Enumeration<String> header = request.getHeaders("X-Forwarded-Proto");
```

Comment:
This header constant is defined in (Jetty) HttpHeader.X_FORWARDED_PROTO (we have a mix of Jakarta and Jetty constants in Whois)

Reply:
I've gone the Jetty way.